flake/dev/generate-all-maintainers.nix

```diff
@@ -0,0 +1,33 @@
+{ lib, ... }:
+{
+  perSystem =
+    { pkgs, config, ... }:
+    {
+      packages.generate-all-maintainers =
+        let
+          meta = pkgs.callPackage ../../stylix/meta.nix { };
+          all_maintainers = lib.pipe meta [
+            lib.attrsToList
+            (map ({ value, ... }: value.maintainers or [ ]))
+            builtins.concatLists
+            builtins.toJSON
+          ];
+        in
+        pkgs.runCommand "generate-all-maintainers" { inherit all_maintainers; } ''
+          echo "// THIS FILE IS GENERATED, DO NOT EDIT" > $out
+          echo "$all_maintainers" >> $out
+        '';
+
+      checks.generated-maintainers =
+        pkgs.runCommand
+          {
+            all_maintainers = ../../stylix/generated/all-maintainers.jsonc;
+            nativeBuildInputs = [ config.packages.generate-all-maintainers ];
```

Comment:
The package doesn't contain any executables, you can't use it as nativeBuildInputs